aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -69,14 +132,54 @@ public void flush() throws IOException {
 
   @Override
   public void write(int b) throws IOException {
+    if (stream.getCount() >= multiPartSize) {
+      newStream();
+      uploadParts();
+    }
```

Comment:
When this is used, the class will never switch to progressive upload. Is that intentional?

Reply:
Yeah, that was just an oversight, it should initiate as well.